src/CozyUtils.js

```diff
@@ -153,44 +154,108 @@ class CozyUtils {
    * @returns {object}
    */
   async computeShortcuts(files) {
+    log('info', 'computeShortcuts starts')
     let schoolShortcuts = []
     let favShortcuts = []
     for (const file of files) {
       if (file.hubMetadata.favori) {
         favShortcuts.push({
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
+          // metadata: {
+          //   icon: svgIcon
+          // },
           shouldReplaceFile: true
         })
       } else {
-        schoolShortcuts.push({
+        const appToSave = {
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
           shouldReplaceFile: true
-        })
+        }
+        if (file.icon) {
+          if (file.icon.match('https://')) {
+            appToSave.metadata = {
+              icon: file.icon
```

Comment:
I'm not sure about those lines. Are-you saying that the `appToSave.metadata.icon` is an URI? If yes, then you have an issue. It should be a base64 encoded string, no?

Reply:
I don't really know, the only thing i can get from the toutatice API (when files have some) is a link to a .png . I know for the manifest (for example) we use .svg images, but there is none in the response from the API, so i was just trying to get something. Also I don't really know how to give a file a specific thumbnail, so i was trying to give them a link, thinking it would be enough. Apparently not, so you should be right :man_shrugging: